### PR TITLE
[BACKLOG-31413] Update "CSV file input" step to use enhanced VFS brow…

### PR DESCRIPTION
**Pull request by ddiroma**
…ser for browsing and selecting files

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 14s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/file-open-save-new/core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1122**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-7055/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by ddiroma** [comment]
@pentaho/r2d2

File: ui/src/main/java/org/pentaho/di/ui/trans/steps/csvinput/CsvInputDialog.java [hunk 1]
```diff
@@ -679,28 +683,11 @@ public void widgetSelected( SelectionEvent se ) {
...
-            BaseMessages.getString( PKG, "System.FileType.AllFiles" ) } );
-
-          if ( dialog.open() != null ) {
-            String str = dialog.getFilterPath() + System.getProperty( "file.separator" ) + dialog.getFileName();
-            wFilename.setText( str );
-          }
-        }
-      } );
+      wbbFilename.addSelectionListener( new SelectionAdapterFileDialogTextVar( log, wFilename, transMeta,
+        new SelectionAdapterOptions( SelectionOperation.FILE,
```
Comment (NJtwentyone) [inline ui/src/main/java/org/pentaho/di/ui/trans/steps/csvinput/CsvInputDialog.java]: no code change necessary, but there is a constructor that accepts FilterType.

`SelectionAdapterOptions( SelectionOperation selectionOperation, FilterType[] filters, FilterType defaultFilter,
                                  String[] providerFilters, boolean useSchemaPath  )`

Comment (ddiroma) [reply ui/src/main/java/org/pentaho/di/ui/trans/steps/csvinput/CsvInputDialog.java]: @NJtwentyone I saw that, but wanted to stay consistent with Corey's implementation of De-serialize.

**Pull request merged**